packages/go_router/example/lib/stateful_nested_navigation.dart

```diff
@@ -0,0 +1,297 @@
+            .toList(),
+        scaffoldBuilder: (BuildContext context, int currentIndex,
+            List<StackedNavigationItemState> itemsState, Widget scaffoldBody) {
+          return ScaffoldWithNavBar(
+              tabs: _tabs,
+              currentIndex: currentIndex,
+              itemsState: itemsState,
+              body: scaffoldBody);
+        },
+
+        /// A transition builder is optional, only included here for
+        /// demonstration purposes.
+        transitionBuilder:
+            (BuildContext context, Animation<double> animation, Widget child) =>
+                FadeTransition(opacity: animation, child: child),
+        routes: <GoRoute>[
+          /// The screen to display as the root in the first tab of the bottom
+          /// navigation bar. Note that the root route must specify the
+          /// `parentNavigatorKey`
+          GoRoute(
+            parentNavigatorKey: _sectionANavigatorKey,
+            path: '/a',
+            builder: (BuildContext context, GoRouterState state) =>
+                const RootScreen(label: 'A', detailsPath: '/a/details'),
+            routes: <RouteBase>[
+              /// The details screen to display stacked on navigator of the
+              /// first tab. This will cover screen A but not the application
+              /// shell (bottom navigation bar).
+              GoRoute(
+                path: 'details',
+                builder: (BuildContext context, GoRouterState state) =>
+                    const DetailsScreen(label: 'A'),
+              ),
+            ],
+          ),
+
+          /// The screen to display as the root in the second tab of the bottom
+          /// navigation bar.
+          GoRoute(
+            parentNavigatorKey: _sectionBNavigatorKey,
+            path: '/b',
+            builder: (BuildContext context, GoRouterState state) =>
+                const RootScreen(
+                    label: 'B',
+                    detailsPath: '/b/details/1',
+                    detailsPath2: '/b/details/2'),
+            routes: <RouteBase>[
+              /// The details screen to display stacked on navigator of the
+              /// second tab. This will cover screen B but not the application
+              /// shell (bottom navigation bar).
+              GoRoute(
+                path: 'details/:param',
+                builder: (BuildContext context, GoRouterState state) =>
+                    DetailsScreen(label: 'B', param: state.params['param']),
+              ),
+            ],
+          ),
+        ],
+      ),
+    ],
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      routeInformationParser: _router.routeInformationParser,
+      routerDelegate: _router.routerDelegate,
+      routeInformationProvider: _router.routeInformationProvider,
```

Comment:
These three lines can be replaced with `routerConfig: _router`.

Reply:
Sweet, thanks